app/src/main/java/com/ericdecanini/usermviapp/UsersPresenter.kt

```diff
@@ -33,15 +33,8 @@ class UsersPresenter {
     }
 
     private fun reduce(previous: UserState?, new: List<User>): UserState {
-        if (previous != null && previous is UserState.DataState) {
```

Comment:
FYI you don't need to do a null check on 'previous' if you're then doing an `is` check on the type.
`if (previous is UserState.DataState)` would have worked without any NPEs